### PR TITLE
Create Post Feature Implementation

### PR DESCRIPTION
**Pull request by nguonodave**
## Description
This PR introduces the complete create post functionality including UI components, form validation, backend API integration, and database migrations.

## Changes
- **UI Components**:
  - Added create post modal with dark/light mode support
  - Implemented privacy selector and image upload with preview

- **Backend**:
  - Created posts model and migration files (including private posts table)
  - Implemented post creation API with image upload support
  - Modified followers API to support 'currentuser' alias

- **Fixes**:
  - Fixed `followsMe` query to only return accepted status
  - Corrected migration command argument checking in the `CreateMigrationFile` function
  - Added proper table existence checks in migrations

### Manual Test Cases

1. **Post Creation Flow**
   - [ ] Open create post modal
   - [ ] Verify all fields are required (title, content, privacy)
   - [ ] Test image upload (JPG/PNG <= 20MB) with preview
   - [ ] Attempt submission with invalid data (should show errors)

2. **Image Upload Validation**
   - [ ] Try uploading non-image files (should reject)
   - [ ] Test files > 20MB (should show size error)
   - [ ] Verify successful upload shows preview

3. **Backend Verification**
   - [ ] Check database after creation:
     - Posts table has new entry
     - Private posts appear in `private_posts` table when applicable
     - Image URLs are properly stored

4. **Edge Cases**
   - [ ] Create post without image (should succeed)
   - [ ] Switch between dark/light mode during creation
   - [ ] Test modal close/cancel behavior

### CODE REVIEW

File: backend/pkg/db/migrations/000002_create_sessions_table.up.sql [hunk 1]
```diff
@@ -1,4 +1,4 @@
-CREATE TABLE sessions (
+CREATE TABLE IF NOT EXISTS sessions (
```
Comment (anamivale) [inline backend/pkg/db/migrations/000002_create_sessions_table.up.sql:1]: I guessed this is why we have down migration?

**Comment by nguonodave** [comment]
> visibility
> * if you assume public has been selected by default  you find an empty string in the backend. please consider having a default visibility
> * created_at time is not okay in the db, both tables.
> * why do we need a private_post table?

- Thanks for catching the bugs, lemme work on them.
- Concerning `private_posts` table, it is essential for tracking which users have access to specific private posts. It acts as a mapping between private posts and the users explicitly granted access.

**Comment by nguonodave** [comment]
> visibility
> * if you assume public has been selected by default  you find an empty string in the backend. please consider having a default visibility
> * created_at time is not okay in the db, both tables.
> * why do we need a private_post table?

Fixed `default visibility` + `created_at` timestamps.
